language/string_spec.rb

```diff
@@ -232,7 +232,8 @@ def long_string_literals
     end
 
     it "produce different objects for literals with the same content in different files if the other file doesn't have the comment" do
-      ruby_exe(fixture(__FILE__, "freeze_magic_comment_across_files_no_comment.rb")).chomp.should == "true"
+      frozen_literals_by_default = eval("'test'").frozen?
+      ruby_exe(fixture(__FILE__, "freeze_magic_comment_across_files_no_comment.rb")).chomp.should == (!frozen_literals_by_default).to_s
     end
```

Comment:
Agreed, best to have two specs and use `guard -> { "str".frozen? } do` or so (which is fine if there is no frozen magic for this file, otherwise needs the `eval`), could you do that?